internal/ingress/controller/store/store.go

```diff
@@ -617,6 +617,11 @@ func New(checkOCSP bool,
 	return store
 }
 
+// CreatesCatchAll returns whether or not an ingress creates a catch-all backend
+func CreatesCatchAll(spec extensions.IngressSpec) bool {
+	return spec.Backend != nil && len(spec.Rules) == 0
```

Comment:
Are we sure there is no case where `len(spec.Rules) > 0` and it's still a catch-all Ingress?

Reply:
That condition is checked at https://github.com/kubernetes/ingress-nginx/blob/master/internal/ingress/controller/controller.go#L952